src/mixins/nr-affiliation-mixin.ts

```diff
@@ -188,8 +188,13 @@ export class NrAffiliationMixin extends Mixins(CommonMixin) {
       // show spinner since this is a network call
       this.$root.$emit('showSpinner', true)
       const accountId = +JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT'))?.id || 0
-      const businessId = await this.createBusinessIA(accountId, legalType)
-      this.goToEntityDashboard(businessId)
+      // incorp requires user login, ignore if user is not logged in and return to NR page
+      if (accountId === 0) {
+        this.$root.$emit('showSpinner', false)
+      } else {
+        const businessId = await this.createBusinessIA(accountId, legalType)
+        this.goToEntityDashboard(businessId)
+      }
```

Comment:
The method executed after user login. When user is not login and click Incorp Now, a login page shows up. If user wants to be back to previous Name Request page at the moment, we expect no error in the case. A short video at the bottom of https://github.com/bcgov/entity/issues/18201 demos the scenario.

Reply:
It is only for Numbered corp name scenario. I don't think there is a NR saved in the case.

Reply:
I think we can raise another ticket if an exact previous page is required to be retrieved when click back button. This ticket is to fix the bug and don't show error page, which is confusing.

Reply:
I just tested the existing NR scenario (don't log in and load up NR 2005711 / ph:1234). Then I clicked Amalgamate Now, then I clicked Back, and it seemed to work correctly.

Reply:
I suppose  you are right -- it would take another ticket if the user wants to get back to the previous page with their data loaded. Right now, in this scenario, the user will have lost their flow and will have to re-enter some data.

Anyway, this still overlaps work done by James. I think it would be better if his PR was merged first and then you updated the new code (in 2 places).

Reply:
OK, your latest changes no longer overlaps with James' work, so I think the PRs can be merged in any order.